static/js/settings_data.ts

```diff
@@ -113,6 +113,24 @@ export function user_can_change_logo(): boolean {
 }
 
 function user_has_permission(policy_value: number): boolean {
+    /* At present, nobody and by_owners_only is not present in
+     * common_policy_values, but we include a check for it here,
+     * so that code using create_web_public_stream_policy_values
+     * or other supersets can use this function. */
+    if (policy_value === settings_config.create_web_public_stream_policy_values.nobody.code) {
+        return false;
+    }
+
+    if (page_params.is_owner) {
+        return true;
+    }
+
+    if (
+        policy_value === settings_config.create_web_public_stream_policy_values.by_owners_only.code
+    ) {
```

Comment:
Now in the same function `user_has_permission` we use values from three different objects `common_message_policy_values`, `create_web_public_stream_policy_values` and `common_policy_values`.

Can we do something like using a same object `policy_values` which will contains all possible values (we can possibl skip description and order fields) and then use this object in this function and use the different objects in the templates. The downside is that this will lead to some more duplication.

Reply:
I think if we combine creating such a shared object, and then also rewrite how e.g. `common_policy_values` is constructed to just pick the 4 keys it allows from that shared object, we'd deduplicate a lot of code and then have this be cleaner as well.